spec/active_interactor/context/base_spec.rb

```diff
@@ -3,33 +3,227 @@
-            subject { context.attributes }
-            it { should be_a Hash }
-            it { should be_empty }
-          end
-
-          describe '`#clean!`' do
-            subject { context.clean! }
-            it { should be_a Hash }
-            it { should be_empty }
-          end
+  after(:each) { described_class.instance_variable_set('@__attributes', []) }
+
+  describe '#attributes' do
+    context 'when no arguments are passed' do
+      subject { described_class.attributes }
+      it { is_expected.to eq [] }
+
+      context 'when an attribute :foo was previously defined' do
+        before { described_class.instance_variable_set('@__attributes', %i[foo]) }
+
+        it { is_expected.to eq %i[foo] }
+      end
+    end
+
+    context 'when given arguments :foo and :bar' do
+      subject { described_class.attributes(:foo, :bar) }
+
+      it { is_expected.to eq %i[bar foo] }
+
+      context 'when an attribute :foo was previously defined' do
+        before { described_class.instance_variable_set('@__attributes', %i[foo]) }
+
+        it { is_expected.to eq %i[bar foo] }
+      end
+    end
+  end
+
+  describe '.attributes' do
+    subject { instance.attributes }
+
+    context 'with class attributes []' do
+      context 'with an instance having attributes { :foo => "foo", :bar => "bar", :baz => "baz" }' do
+        let(:instance) { described_class.new(foo: 'foo', bar: 'bar', baz: 'baz') }
+
+        it { is_expected.to be_a Hash }
+        it { is_expected.to be_empty }
+      end
+    end
+
+    context 'with class attributes [:foo, :bar, :baz]' do
+      before { described_class.attributes(:foo, :bar, :baz) }
+
+      context 'with an instance having attributes { :foo => "foo", :bar => "bar", :baz => "baz" }' do
+        let(:instance) { described_class.new(foo: 'foo', bar: 'bar', baz: 'baz') }
+
+        it { is_expected.to be_a Hash }
+        it { is_expected.to eq(bar: 'bar', baz: 'baz', foo: 'foo') }
+      end
+    end
+  end
+
+  describe '#called!' do
+    subject do
+      instance.called!(interactor1)
+      instance.called!(interactor2)
+    end
+
+    let(:instance) { described_class.new }
+    let(:interactor1) { double(:interactor1) }
+    let(:interactor2) { double(:interactor2) }
+
+    it 'is expected to append interactors to instance variable _called' do
+      expect { subject }.to change { instance.send(:_called) }
+        .from([])
+        .to([interactor1, interactor2])
+    end
+  end
+
+  describe '#fail!' do
+    subject { instance.fail!(errors) }
+    let(:instance) { described_class.new }
+
+    context 'with errors equal to nil' do
+      let(:errors) { nil }
+
+      it { expect { subject }.to raise_error(ActiveInteractor::Error::ContextFailure) }
+
+      it 'is expected not to merge errors' do
+        expect(instance.errors).not_to receive(:merge!)
+        subject
+      rescue ActiveInteractor::Error::ContextFailure # rubocop:disable Lint/SuppressedException
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress exceptions.](https://app.codacy.com/manual/aaronmallen/activeinteractor/pullRequest?prid=4767422)